devito/operations/interpolators.py

```diff
@@ -382,7 +382,10 @@ def _interpolate(self, expr, increment=False, self_subs={}, implicit_dims=None):
                                            subdomain=subdomain)
 
         # Accumulate point-wise contributions into a temporary
-        rhs = Symbol(name='sum', dtype=self.sfunction.dtype)
+        # Use unique name per SparseFunction to avoid variable reuse across
```

Comment:
Is this actually the issue? It doesn't seem to be from the generated code...
`sum` is always initialised as:
```C
float sum = 0.0F;
```
at the start of each interpolation loop and the interpolation loops don't get fused

Reply:
Misdiagnosis. Reverted.